### PR TITLE
feat: serde trait for field and group elements

### PR DESCRIPTION
**Pull request by zhenfeizhang**
`serde` is quite useful for downstream libraries. would love to have this feature enabled
 
the user can enable it with `feature = derive_serde`

### CODE REVIEW

**Review by han0110 — APPROVED** [review]
LGTM! But I remember the reason to manually implement `SerdeObject` is to avoid dependency `serde`, so I think it'd be nice to have flag to enable/disable this.

File: Cargo.toml [hunk 1]
```diff
@@ -34,6 +35,7 @@ lazy_static = "1.4.0"
 num-bigint = "0.4.3"
 num-traits = "0.2"
 paste = "1.0.11"
+serde = { version = "1.0", default-features = false, features = ["derive"] }
```
Comment (CPerezz) [inline Cargo.toml:38]: I'd create a serde feature and import this under it.

**Review by CPerezz — CHANGES_REQUESTED** [review]
Please, put this behind a `serde` feature. The derives and the imports as well as the serde dependency. 

Users that don't need it should not pay for it in compilation time.

**Comment by zhenfeizhang** [comment]
cool. added in 5b97d1182e855fffe6e22a25aec2f6f7fc8befe0

**Comment by zhenfeizhang** [comment]
Don't know why CI failed... 
```
 cargo test --verbose --release --all --features asm
 ```
 passed on my local machine

**Comment by CPerezz** [comment]
> Don't know why CI failed...
> 
> ```
>  cargo test --verbose --release --all --features asm
> ```
> 
> passed on my local machine

No worries. It happens sometimes. Not sure what's the exact issue with the gh-runner TBH.
I triggered it again. Let's hope it works now.

**Review by CPerezz — APPROVED** [review]
LGTM! Thanks for adding the feature!!

**Comment by han0110** [comment]
Hi @zhenfeizhang is there any specific reason not re-export `ff` and `group` by default? I think it'd a little awkward for downstream libraries since all of those relying on `halo2curves` now need to specify the `reexport` feature (I think most of them need to import the traits in `ff` and `group`).

If no specific concern I'd suggest to make re-export it by default, what do you think?

**Pull request merged**